### ISSUE TITLE
Deepin screen recorder coredump / glitchy files

### ISSUE THREAD

**Issue by keybreak** [linuxdeepin/developer-center#940]
## Describe the bug
Deepin screen recorder throws coredump & produces glitchy super-small file.

## To Reproduce
1. Open Deepin screen recorder, select part of screen or fullscreen in mp4 mode
2. Open Deepin file manager, for example etc folder, try to scroll with touchpad

## Expected behavior
No coredumps / crashes / good files :)

## Enviroments:
OS: Manjaro 18.0.3 Illyria
Kernel: x86_64 Linux 4.19.24-1-MANJARO
Shell: bash 5.0.0
DE: Deepin 15.9
WM: Deepin WM
WM Theme: deepin
GTK Theme: deepin [GTK2/3]
Icon Theme: deepin
bits: 64
compiler: gcc 8.2.1
tk: Qt 5.12.1
dm: LightDM

### Distro & Version
Manjaro Deepin 18.0.3

### Related package version
deepin-file-manager 1:4.7.6-2
deepin-screen-recorder 2.7.7-1

### Additional context
Seems to happen only on my real test system 1, VM with same Manjaro Deepin doesn't have this...
This is started only after 23.02.2019 updates, before everything was ok.

Tested with mouse, it's same, so i doubt it's touchpad related.

**inxi -Fxxz**
```
System:    Host: x133-pc Kernel: 4.19.24-1-MANJARO x86_64 bits: 64 compiler: gcc v: 8.2.1 Desktop: Deepin 15.9 
           tk: Qt 5.12.1 wm: DeepinWM dm: LightDM Distro: Manjaro Linux 
Machine:   Type: Laptop System: Sony product: VGN-Z11VRN_B v: J003DTPY serial: <filter> Chassis: type: 10 
           serial: <filter> 
           Mobo: Sony model: VAIO serial: <filter> BIOS: INSYDE v: R5031M3 date: 11/10/2009 
Battery:   ID-1: BAT0 charge: 33.9 Wh condition: 33.9/59.5 Wh (57%) volts: 12.3/10.8 model: Sony Corporation serial: N/A 
           status: Full 
CPU:       Topology: Dual Core model: Intel Core2 Duo P9500 bits: 64 type: MCP arch: Penryn rev: 6 L2 cache: 6144 KiB 
           flags: lm nx pae sse sse2 sse3 sse4_1 ssse3 vmx bogomips: 10113 
           Speed: 2156 MHz min/max: 1600/2533 MHz Core speeds (MHz): 1: 1596 2: 1596 
Graphics:  Device-1: Intel Mobile 4 Series Integrated Graphics vendor: Sony driver: i915 v: kernel bus ID: 00:02.0 
           chip ID: 8086:2a42 
           Display: x11 server: X.Org 1.20.3 driver: intel unloaded: modesetting alternate: fbdev,vesa 
           resolution: 1600x900~60Hz 
           OpenGL: renderer: Mesa DRI Mobile Intel GM45 Express v: 2.1 Mesa 18.3.3 direct render: Yes 
Audio:     Device-1: Intel 82801I HD Audio vendor: Sony driver: snd_hda_intel v: kernel bus ID: 00:1b.0 
           chip ID: 8086:293e 
           Sound Server: ALSA v: k4.19.24-1-MANJARO 
Network:   Device-1: Intel 82567LM Gigabit Network vendor: Sony driver: e1000e v: 3.2.6-k port: 7100 bus ID: 00:19.0 
           chip ID: 8086:10f5 
           IF: enp0s25 state: down mac: <filter> 
           Device-2: Qualcomm Atheros AR928X Wireless Network Adapter vendor: Foxconn driver: ath9k v: kernel port: 7000 
           bus ID: 06:00.0 chip ID: 168c:002a 
           IF: wlp6s0 state: up mac: <filter> 
Drives:    Local Storage: total: 1.36 TiB used: 145.85 GiB (10.4%) 
           ID-1: /dev/sda vendor: HGST (Hitachi) model: HTS541515A9E630 size: 1.36 TiB speed: 3.0 Gb/s serial: <filter> 
Partition: ID-1: / size: 172.02 GiB used: 11.02 GiB (6.4%) fs: ext4 dev: /dev/sda1 
Sensors:   System Temperatures: cpu: 45.0 C mobo: N/A 
           Fan Speeds (RPM): N/A 
Info:      Processes: 150 Uptime: 13m Memory: 3.78 GiB used: 623.9 MiB (16.1%) Init: systemd v: 241 Compilers: 
           gcc: 8.2.1 Shell: bash v: 5.0.0 running in: deepin-terminal inxi: 3.0.30 
```

Here's coredump **sudo journalctl -p3 -xb**:
```
feb 23 15:40:33 x133-pc daemon/dde-system-daemon[901]: main.go:89: failed to request name: name com.deepin.daemon.Daemon already taken
                                                            ->  proc.go:201
                                                            ->  asm_amd64.s:1333
feb 23 15:57:57 x133-pc deepin-screen-recorder[2333]: QThread: Destroyed while thread is still running
feb 23 15:57:58 x133-pc systemd-coredump[2359]: Process 2333 (deepin-screen-r) of user 1000 dumped core.
                                                   
                                                   Stack trace of thread 2333:
                                                   #0  0x00007f1caa0cfd7f raise (libc.so.6)
                                                   #1  0x00007f1caa0ba672 abort (libc.so.6)
                                                   #2  0x00007f1caa48b7fc _ZNK14QMessageLogger5fatalEPKcz (libQt5Core.so.5)
                                                   #3  0x00007f1caa48c055 n/a (libQt5Core.so.5)
                                                   #4  0x000055d9ac66d283 n/a (deepin-screen-recorder)
                                                   #5  0x000055d9ac66c840 main (deepin-screen-recorder)
                                                   #6  0x00007f1caa0bc223 __libc_start_main (libc.so.6)
                                                   #7  0x000055d9ac66cb1e n/a (deepin-screen-recorder)
                                                   
                                                   Stack trace of thread 2334:
                                                   #0  0x00007f1caa188c21 __poll (libc.so.6)
                                                   #1  0x00007f1caa07a630 n/a (libxcb.so.1)
                                                   #2  0x00007f1caa07c2db xcb_wait_for_event (libxcb.so.1)
                                                   #3  0x00007f1ca427d949 n/a (libQt5XcbQpa.so.5)
                                                   #4  0x00007f1caa4c596c n/a (libQt5Core.so.5)
                                                   #5  0x00007f1ca8ecaa9d start_thread (libpthread.so.0)
                                                   #6  0x00007f1caa193b23 __clone (libc.so.6)
                                                   
                                                   Stack trace of thread 2336:
                                                   #0  0x00007f1caa188c21 __poll (libc.so.6)
                                                   #1  0x00007f1ca9978540 n/a (libglib-2.0.so.0)
                                                   #2  0x00007f1ca997862e g_main_context_iteration (libglib-2.0.so.0)
                                                   #3  0x00007f1cabedcbde n/a (libdconfsettings.so)
                                                   #4  0x00007f1ca995367b n/a (libglib-2.0.so.0)
                                                   #5  0x00007f1ca8ecaa9d start_thread (libpthread.so.0)
                                                   #6  0x00007f1caa193b23 __clone (libc.so.6)
                                                   
                                                   Stack trace of thread 2338:
                                                   #0  0x00007f1caa188c21 __poll (libc.so.6)
                                                   #1  0x00007f1ca9978540 n/a (libglib-2.0.so.0)
                                                   #2  0x00007f1ca99795c2 g_main_loop_run (libglib-2.0.so.0)
                                                   #3  0x00007f1ca7e8ccb8 n/a (libgio-2.0.so.0)
                                                   #4  0x00007f1ca995367b n/a (libglib-2.0.so.0)
                                                   #5  0x00007f1ca8ecaa9d start_thread (libpthread.so.0)
                                                   #6  0x00007f1caa193b23 __clone (libc.so.6)
                                                   
                                                   Stack trace of thread 2340:
                                                   #0  0x00007f1caa188c21 __poll (libc.so.6)
                                                   #1  0x00007f1caa07a630 n/a (libxcb.so.1)
                                                   #2  0x00007f1caa07bfef n/a (libxcb.so.1)
                                                   #3  0x00007f1caa07c162 xcb_wait_for_reply64 (libxcb.so.1)
                                                   #4  0x00007f1cabdc22d9 _XReply (libX11.so.6)
                                                   #5  0x00007f1cabb6964e XRecordEnableContext (libXtst.so.6)
                                                   #6  0x000055d9ac68064d n/a (deepin-screen-recorder)
                                                   #7  0x00007f1caa4c596c n/a (libQt5Core.so.5)
                                                   #8  0x00007f1ca8ecaa9d start_thread (libpthread.so.0)
                                                   #9  0x00007f1caa193b23 __clone (libc.so.6)
                                                   
                                                   Stack trace of thread 2339:
                                                   #0  0x00007f1ca8ed0afc pthread_cond_wait@@GLIBC_2.3.2 (libpthread.so.0)
                                                   #1  0x00007f1c8b7fec84 n/a (i965_dri.so)
                                                   #2  0x00007f1c8b7fe9a8 n/a (i965_dri.so)
                                                   #3  0x00007f1ca8ecaa9d start_thread (libpthread.so.0)
                                                   #4  0x00007f1caa193b23 __clone (libc.so.6)
                                                   
                                                   Stack trace of thread 2335:
                                                   #0  0x00007f1caa188c21 __poll (libc.so.6)
                                                   #1  0x00007f1ca9978540 n/a (libglib-2.0.so.0)
                                                   #2  0x00007f1ca997862e g_main_context_iteration (libglib-2.0.so.0)
                                                   #3  0x00007f1ca9978682 n/a (libglib-2.0.so.0)
                                                   #4  0x00007f1ca995367b n/a (libglib-2.0.so.0)
                                                   #5  0x00007f1ca8ecaa9d start_thread (libpthread.so.0)
                                                   #6  0x00007f1caa193b23 __clone (libc.so.6)
                                                   
                                                   Stack trace of thread 2337:
                                                   #0  0x00007f1caa188c21 __poll (libc.so.6)
                                                   #1  0x00007f1ca9978540 n/a (libglib-2.0.so.0)
                                                   #2  0x00007f1ca997862e g_main_context_iteration (libglib-2.0.so.0)
                                                   #3  0x00007f1caa6d7ce9 _ZN20QEventDispatcherGlib13processEventsE6QFlagsIN10QEventLoop17ProcessEventsFlagEE (libQt5Core.so.5)
                                                   #4  0x00007f1caa680b2c _ZN10QEventLoop4execE6QFlagsINS_17ProcessEventsFlagEE (libQt5Core.so.5)
                                                   #5  0x00007f1caa4c4569 _ZN7QThread4execEv (libQt5Core.so.5)
                                                   #6  0x00007f1caa94bba6 n/a (libQt5DBus.so.5)
                                                   #7  0x00007f1caa4c596c n/a (libQt5Core.so.5)
                                                   #8  0x00007f1ca8ecaa9d start_thread (libpthread.so.0)
                                                   #9  0x00007f1caa193b23 __clone (libc.so.6)
-- Subject: Process 2333 (deepin-screen-r) dumped core
-- Defined-By: systemd
-- Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
-- Documentation: man:core(5)
-- 
-- Process 2333 (deepin-screen-r) crashed and dumped core.
-- 
-- This usually indicates a programming error in the crashing program and
-- should be reported to its vendor as a bug.
```

**Comment by keybreak** [comment]
It;'s easily reproduceable on my system, so if i can get any more information - please let me know how.

**Comment by felixonmars** [comment]
The segfault has been there since the very beginning, and it shouldn't be affecting the resulting video.

I cannot fully reproduce the "produces glitchy super-small file" part though. In GIF mode the result seems correct, and in mp4 mode the result is glitchy but not super small here.

**Comment by keybreak** [comment]
@felixonmars
I've never ever seen this before today with both of my real test machines and VM (and believe me, i would notice this), not on Deepin Linux, not on Arch, not on Manjaro Deepin.

Besides, remember few days ago i've recorded this?
https://i.imgur.com/LLI13yW.mp4

I couldn't possibly do this, if it would happen before, because now if i'll just press record and try to open for example Deepin file mananger, output file will show just mouse moving on wallpaper (without actually showing folders / files on desktop or Deepin file manager) - that's what i meant by glitchy

**Comment by felixonmars** [comment]
Well, I saw the crash before and for many times, since I packaged it the first time, and upstream doesn't really care about a crash on exit. For me it has nothing to do with a broken record result. Please try to get some ffmpeg output from stdout/stderr of the program if you would like to diagnose that problem.

**Comment by keybreak** [comment]
@felixonmars
Sure, but i'm not familiar with this concepts yet, can you please give step by step guide of how to properly get it?

**Comment by felixonmars** [comment]
Well, to get stdout/stderr of a program, you need to run it from a terminal.

**Comment by keybreak** [comment]
Oh! :rofl: 
Ok now i know what it means, thank you :)

1. This time output video have recorded desktop, it's icons, terminal, but not Deepin file manager
2. **Error** and everything after it started from the moment i've opened Deepin file manager

```
translatePath after feedback: "/usr/share/dtkwidget/translations/dtkwidget2_ru"
"deepin-screen-recorder" can not find qm files
requestActivate() called for  QWidgetWindow(0x56221c68cb30, name="MainWindowClassWindow")  which has Qt::WindowDoesNotAcceptFocus set.
Not found mp4_framerate option in config file, mp4 use framerate 25
mp4 framerate  25
startup notification initiated:  startdde-567-x133-pc-dde-file-manager-0_TIME79439
startup notification completed:  startdde-567-x133-pc-dde-file-manager-0_TIME79439
Error
"ffmpeg version n4.1.1 Copyright (c) 2000-2019 the FFmpeg developers"
"  built with gcc 8.2.1 (GCC) 20181127"
"  configuration: --prefix=/usr --disable-debug --disable-static --disable-stripping --enable-fontconfig --enable-gmp --enable-gnutls --enable-gpl --enable-ladspa --enable-libaom --enable-libass --enable-libbluray --enable-libdrm --enable-libfreetype --enable-libfribidi --enable-libgsm --enable-libiec61883 --enable-libjack --enable-libmodplug --enable-libmp3lame --enable-libopencore_amrnb --enable-libopencore_amrwb --enable-libopenjpeg --enable-libopus --enable-libpulse --enable-libsoxr --enable-libspeex --enable-libssh --enable-libtheora --enable-libv4l2 --enable-libvidstab --enable-libvorbis --enable-libvpx --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxcb --enable-libxml2 --enable-libxvid --enable-nvdec --enable-nvenc --enable-omx --enable-shared --enable-version3"
"  libavutil      56. 22.100 / 56. 22.100"
"  libavcodec     58. 35.100 / 58. 35.100"
"  libavformat    58. 20.100 / 58. 20.100"
"  libavdevice    58.  5.100 / 58.  5.100"
"  libavfilter     7. 40.101 /  7. 40.101"
"  libswscale      5.  3.100 /  5.  3.100"
"  libswresample   3.  3.100 /  3.  3.100"
"  libpostproc    55.  3.100 / 55.  3.100"
"[x11grab @ 0x55e8099a6d80] Stream #0: not enough frames to estimate rate; consider increasing probesize"
"Input #0, x11grab, from ':0+0,0':"
"  Duration: N/A, start: 1550933980.670019, bitrate: N/A"
"    Stream #0:0: Video: rawvideo (BGR[0] / 0x524742), bgr0, 1600x900, 25 fps, 1000k tbr, 1000k tbn, 1000k tbc"
"Stream mapping:"
"  Stream #0:0 -> #0:0 (rawvideo (native) -> h264 (libx264))"
"Press [q] to stop, [?] for help"
"[libx264 @ 0x55e8099b1e40] using cpu capabilities: MMX2 SSE2Fast SSSE3 SSE4.1 Cache64"
"[libx264 @ 0x55e8099b1e40] profile High, level 4.0"
"[libx264 @ 0x55e8099b1e40] 264 - core 155 r2917 0a84d98 - H.264/MPEG-4 AVC codec - Copyleft 2003-2018 - http://www.videolan.org/x264.html - options: cabac=1 ref=3 deblock=1:0:0 analyse=0x3:0x113 me=hex subme=7 psy=1 psy_rd=1.00:0.00 mixed_ref=1 me_range=16 chroma_me=1 trellis=1 8x8dct=1 cqm=0 deadzone=21,11 fast_pskip=1 chroma_qp_offset=-2 threads=3 lookahead_threads=1 sliced_threads=0 nr=0 decimate=1 interlaced=0 bluray_compat=0 constrained_intra=0 bframes=3 b_pyramid=2 b_adapt=1 b_bias=0 direct=1 weightb=1 open_gop=0 weightp=2 keyint=250 keyint_min=25 scenecut=40 intra_refresh=0 rc_lookahead=40 rc=crf mbtree=1 crf=23.0 qcomp=0.60 qpmin=0 qpmax=69 qpstep=4 ip_ratio=1.40 aq=1:1.00"
"Output #0, mp4, to '/tmp/deepin-screen-recorder_Desktop_20190223175938.mp4':"
"  Metadata:"
"    encoder         : Lavf58.20.100"
"    Stream #0:0: Video: h264 (libx264) (avc1 / 0x31637661), yuv420p, 1600x900, q=-1--1, 25 fps, 12800 tbn, 25 tbc"
"    Metadata:"
"      encoder         : Lavc58.35.100 libx264"
"    Side data:"
"      cpb: bitrate max/min/avg: 0/0/0 buffer size: 0 vbv_delay: -1"
"frame=   18 fps=0.0 q=0.0 size=       0kB time=00:00:00.00 bitrate=N/A dup=15 drop=0 speed=   0x    \rframe=   33 fps= 31 q=0.0 size=       0kB time=00:00:00.00 bitrate=N/A dup=15 drop=9 speed=   0x    \rframe=   47 fps= 28 q=0.0 size=       0kB time=00:00:00.00 bitrate=N/A dup=15 drop=13 speed=   0x    \rframe=   60 fps= 27 q=28.0 size=       0kB time=00:00:00.40 bitrate=   1.0kbits/s dup=21 drop=13 speed=0.179x    \rframe=   73 fps= 27 q=28.0 size=       0kB time=00:00:00.92 bitrate=   0.4kbits/s dup=27 drop=13 speed=0.335x    \rframe=   87 fps= 26 q=28.0 size=       0kB time=00:00:01.48 bitrate=   0.3kbits/s dup=32 drop=13 speed=0.45x    \rframe=   99 fps= 26 q=28.0 size=       0kB time=00:00:01.96 bitrate=   0.2kbits/s dup=35 drop=13 speed=0.516x    \rframe=  113 fps= 26 q=28.0 size=       0kB time=00:00:02.52 bitrate=   0.2kbits/s dup=41 drop=13 speed=0.577x    \rframe=  128 fps= 26 q=28.0 size=       0kB time=00:00:03.12 bitrate=   0.1kbits/s dup=52 drop=13 speed=0.63x    \rframe=  142 fps= 26 q=28.0 size=       0kB time=00:00:03.68 bitrate=   0.1kbits/s dup=63 drop=13 speed=0.67x    \rframe=  157 fps= 26 q=28.0 size=       0kB time=00:00:04.28 bitrate=   0.1kbits/s dup=76 drop=13 speed=0.702x    \rframe=  179 fps= 26 q=28.0 size=       0kB time=00:00:05.16 bitrate=   0.1kbits/s dup=96 drop=13 speed=0.738x    \rframe=  195 fps= 26 q=28.0 size=       0kB time=00:00:05.80 bitrate=   0.1kbits/s dup=111 drop=13 speed=0.761x    \rframe=  209 fps= 26 q=28.0 size=       0kB time=00:00:06.36 bitrate=   0.1kbits/s dup=124 drop=13 speed=0.778x    \rframe=  232 fps= 25 q=28.0 size=       0kB time=00:00:07.28 bitrate=   0.1kbits/s dup=145 drop=13 speed=0.799x    \rframe=  251 fps= 25 q=28.0 size=       0kB time=00:00:08.04 bitrate=   0.0kbits/s dup=162 drop=13 speed=0.815x    \rframe=  274 fps= 25 q=28.0 size=       0kB time=00:00:08.96 bitrate=   0.0kbits/s dup=183 drop=13 speed=0.83x    \rframe=  287 fps= 25 q=28.0 size=       0kB time=00:00:09.48 bitrate=   0.0kbits/s dup=195 drop=13 speed=0.839x    \rframe=  311 fps= 25 q=28.0 size=     256kB time=00:00:10.44 bitrate= 200.9kbits/s dup=217 drop=13 speed=0.851x    \rframe=  338 fps= 25 q=28.0 size=     256kB time=00:00:11.52 bitrate= 182.1kbits/s dup=242 drop=13 speed=0.863x    \rframe=  357 fps= 25 q=28.0 size=     256kB time=00:00:12.28 bitrate= 170.8kbits/s dup=260 drop=13 speed=0.87x    \rframe=  379 fps= 25 q=28.0 size=     256kB time=00:00:13.16 bitrate= 159.4kbits/s dup=281 drop=13 speed=0.878x    \rframe=  406 fps= 25 q=28.0 size=     256kB time=00:00:14.24 bitrate= 147.3kbits/s dup=307 drop=13 speed=0.886x    \rframe=  429 fps= 25 q=28.0 size=     256kB time=00:00:15.16 bitrate= 138.4kbits/s dup=329 drop=13 speed=0.893x    \rframe=  448 fps= 25 q=28.0 size=     256kB time=00:00:15.92 bitrate= 131.8kbits/s dup=347 drop=13 speed=0.896x    \rframe=  463 fps= 25 q=28.0 size=     256kB time=00:00:16.52 bitrate= 127.0kbits/s dup=361 drop=13 speed=0.899x    \rframe=  484 fps= 25 q=28.0 size=     256kB time=00:00:17.36 bitrate= 120.8kbits/s dup=380 drop=13 speed=0.905x    \rframe=  498 fps= 25 q=28.0 size=     256kB time=00:00:17.92 bitrate= 117.0kbits/s dup=392 drop=13 speed=0.907x    \rframe=  513 fps= 25 q=28.0 size=     256kB time=00:00:18.52 bitrate= 113.3kbits/s dup=403 drop=13 speed=0.911x    \rframe=  526 fps= 25 q=28.0 size=     256kB time=00:00:19.04 bitrate= 110.2kbits/s dup=408 drop=13 speed=0.913x    \rframe=  533 fps= 24 q=-1.0 Lsize=     423kB time=00:00:21.20 bitrate= 163.5kbits/s dup=408 drop=13 speed=0.943x    "
"video:416kB audio:0kB subtitle:0kB other streams:0kB global headers:0kB muxing overhead: 1.695902%"
"[libx264 @ 0x55e8099b1e40] frame I:3     Avg QP:12.48  size:113732"
"[libx264 @ 0x55e8099b1e40] frame P:134   Avg QP:13.77  size:   390"
"[libx264 @ 0x55e8099b1e40] frame B:396   Avg QP:22.38  size:    80"
"[libx264 @ 0x55e8099b1e40] consecutive B-frames:  0.9%  0.0%  0.0% 99.1%"
"[libx264 @ 0x55e8099b1e40] mb I  I16..4: 16.3% 63.8% 19.8%"
"[libx264 @ 0x55e8099b1e40] mb P  I16..4:  0.0%  0.0%  0.0%  P16..4:  1.5%  0.2%  0.2%  0.0%  0.0%    skip:98.2%"
"[libx264 @ 0x55e8099b1e40] mb B  I16..4:  0.0%  0.0%  0.0%  B16..8:  0.5%  0.0%  0.0%  direct: 0.0%  skip:99.5%  L0:54.5% L1:45.5% BI: 0.0%"
"[libx264 @ 0x55e8099b1e40] 8x8 transform intra:63.0% inter:97.3%"
"[libx264 @ 0x55e8099b1e40] coded y,uvDC,uvAC intra: 61.3% 88.4% 81.1% inter: 0.1% 0.2% 0.0%"
"[libx264 @ 0x55e8099b1e40] i16 v,h,dc,p: 15% 29%  4% 53%"
"[libx264 @ 0x55e8099b1e40] i8 v,h,dc,ddl,ddr,vr,hd,vl,hu: 33% 21% 13%  7%  5%  4%  4%  6%  7%"
"[libx264 @ 0x55e8099b1e40] i4 v,h,dc,ddl,ddr,vr,hd,vl,hu: 39% 28%  9%  4%  5%  4%  4%  3%  3%"
"[libx264 @ 0x55e8099b1e40] i8c dc,h,v,p: 42% 25% 14% 19%"
"[libx264 @ 0x55e8099b1e40] Weighted P-Frames: Y:0.0% UV:0.0%"
"[libx264 @ 0x55e8099b1e40] ref P L0: 91.2%  0.1%  7.5%  1.2%"
"[libx264 @ 0x55e8099b1e40] ref B L0: 14.9% 85.0%  0.0%"
"[libx264 @ 0x55e8099b1e40] ref B L1: 95.1%  4.9%"
"[libx264 @ 0x55e8099b1e40] kb/s:159.60"
"Exiting normally, received signal 15."
""
QThread: Destroyed while thread is still running

```

**Comment by felixonmars** [comment]
I see. Since I can also get the glitch and there is actually no errors in the ffmpeg logs I believe it's a bug somewhere in ffmpeg's x11grab and/or X, or the video drivers. Deepin screen recorder is just a ui for that :/

**Comment by keybreak** [comment]
I've noticed Manjaro had some X related packages updated today like few hours ago, @oberon-manjaro any thoughts on that?

**Comment by keybreak** [comment]
I was trying to narrow it down more, but i don't undestand why it doesn't happen on VM, and happens on real machine.

**Comment by SiamSami** [comment]
Have you tried reinstalling the whole desktop environment?

**Comment by keybreak** [comment]
Not really, but i guess it would be too overkill to consider, since @felixonmars also have it and most likely a lot of Arch / Manjaro Deepin users meeting some of conditions except X updates, which yet to be determined...

**Comment by keybreak** [comment]
@felixonmars 
By now still same situation, do you have any definitive clue of what is the cause?

**Comment by keybreak** [comment]
@BLumia @hualet 
Can you please look into it?
Since 23 february on Intel videocards Deepin screen recorder is unusable, that's not good.

**Comment by hualet** [comment]
@keybreak what if you use another Window Manager?

**Comment by keybreak** [comment]
@hualet 
Tested with Pcmanfm-qt - yes.

Also i've noticed that:
1. Right now basically it's not connected to file-manager at all, if you will capture whole screen regardless of what you're trying to do and which program is opened you'll get desktop + moving cursor and if lucky some jerky movement of program (but not likely even that)...

2. When such crash occurs in session first time, next time you'll try to capture it will produce exactly same broken file (with same content i mean), even if you deleted it already.

____
### Enviroments

OS: Manjaro 18.0.4 Illyria
Kernel: x86_64 Linux 4.19.34-1-MANJARO
Shell: bash
DE: Deepin 15.10
WM: Deepin WM
WM Theme: deepin
GTK Theme: deepin [GTK2/3]
Icon Theme: deepin
compiler: gcc 8.2.1
tk: Qt 5.12.2
dm: LightDM

Machine specs are same as in OP

**Comment by hualet** [comment]
@keybreak , I mean Window Manager, not File Manager :sweat_smile:

**Comment by hualet** [comment]
We did notice there're some glitches now and then if the window manager is deepin-wm/mutter while developing `miraclecast`.

**Comment by keybreak** [comment]
@hualet 
My bad, need to get some sleep to read properly! :laughing: 
Sorry, i can't test another WM on this particular machine right now...

If you have such suspicions, let's wait when deepin-kwin become default wm then repeat test, to make sure & not waste your time. :smile_cat:

**Comment by justforlxz** [comment]
Sorry, this issue will be closed soon. If it is necessary to discuss it again, please create a new issue.
